Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts

```diff
@@ -485,6 +488,35 @@ namespace gdjs {
       point[1] = position.y;
       return point;
     }
+    updateAntialiasing(): void {
+      if (this._object.getAntialiasing() !== 'None') {
+        const antialiasingFilter =
+          this._antialiasingFilter ||
+          (this._antialiasingFilter = new PIXI.filters.FXAAFilter());
+        antialiasingFilter.enabled = true;
+        antialiasingFilter.multisample =
+          PIXI.MSAA_QUALITY[this._object.getAntialiasing().toUpperCase()] ||
+          PIXI.MSAA_QUALITY.LOW;
+
+        if (!this._graphics.filters) {
+          this._graphics.filters = [];
+        }
+        // Do not apply the filter if it is already present on the object.
+        if (this._graphics.filters.indexOf(this._antialiasingFilter) === -1) {
+          this._graphics.filters.push(antialiasingFilter);
+        }
+      } else if (this._antialiasingFilter !== null) {
+        let antialiasingFilterIndex = this._graphics.filters?.indexOf(
```

Comment:
This is wrong - you might be confusing it with `!.`. `!.` is the typescript non-null type cast operator, which tells tyopescript to treat a `T | null` type as if it was `T`. `?.` is an ecmascript expression for optional chaining - if it exist, continue the operation, else return undefined. The expression return value is then correctly typed as `number | undefined` as a result. As you can see just after, the case where it is not defined is handled with `antialisaisingFilrerIndex !== undefined`.

Reply:
I'm not saying this is not a valid operator in this case. I'm saying we are not using it in the codebase so for consistency I'm removing it too.
It's a coding preference to avoid using it (personal too). Yes, it makes the codebase more verbose, but forces the developer to think at what they're doing.
"is this not defined? why? should I handle it before? return early?" etc...

Reply:
Ah I see, excuse me for the misunderstanding.